lib/Raven/Serializer.php

```diff
@@ -109,6 +109,12 @@ protected function serializeValue($value)
         if (is_null($value) || is_bool($value) || is_float($value) || is_integer($value)) {
             return $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
+            // If the value is a callable and expects no parameters, we evaluate the
+            // callable value by calling it.
+            if (is_callable($value) && Raven_Util::getCallableParamNum($value) === 0) {
+                return $value();
```

Comment:
@ste93cry you make a fair point. In my opinion those are more of issues for the implementor to tackle with. I do understand if it would create an unwanted situation where the SDK is blamed for something that _looks_ like the SDKs problem (as in devs make bad decisions and then decide the SDK is bad for allowing it, instead of taking the time to see what the actual problem is).

Error situations are hairy. I tried a few times and could read global state properly via framework functions during a simulated error but I guess there are situations where all is lost. I wonder if PHP7s Throwables could be used to some extent when evaluating callables in case errors happen during evaluation and the serialization is stopped from that point forward for the callable. Hard fatal errors are a bit tricky though.